xgcm/grid.py

```diff
@@ -1667,10 +1700,53 @@ def _1d_grid_ufunc_dispatch(
 
         if isinstance(axis, str):
             axis = [axis]
-        if to is None:
-            to = {ax: None for ax in axis}
-        elif isinstance(to, str):
-            to = {ax: to for ax in axis}
+
+        # convert inpug arguments into axes-kwarg mappings
+        to = self._as_axis_kwarg_mapping(to)
+        # if to is None:
+        #     to = {ax: None for ax in axis}
+        # elif isinstance(to, str):
+        #     to = {ax: to for ax in axis}
+
+        # If I understand correctly, this could contain some other kwargs, which are not appropriate to convert
+        # Note that the option to pass boundary (actually padding) options on the method will be deprecated,
+        # so for now I will just store everything I need here.
+        VALID_BOUNDARY_KWARGS = ["boundary", "periodic", "fill_value"]
+        kwargs = {
+            k: self._as_axis_kwarg_mapping(v) if k in VALID_BOUNDARY_KWARGS else v
+            for k, v in kwargs.items()
+        }
+        # Now check if some of the values not provided in the kwargs are set as grid properties
+        # In the future this should be the only way to access that information!
+        kwargs_with_defaults = {}
+
+        for k in np.unique(
+            list(kwargs.keys()) + VALID_BOUNDARY_KWARGS
```

Comment:
Can't you use `set()` instead of `np.unique()`?

Reply:
Um I guess. Let me try.

Reply:
Yup seems to work.

Reply:
Thanks - using set is much more pythonic, less error-prone, and I expect a lot quicker.